packages/grid/x-data-grid/src/hooks/features/keyboardNavigation/useGridKeyboardNavigation.ts

```diff
@@ -134,15 +138,27 @@ export const useGridKeyboardNavigation = (
         }
 
         case 'ArrowRight': {
-          if (colIndexBefore < lastColIndex) {
-            goToCell(colIndexBefore + 1, getRowIdFromIndex(rowIndexBefore), 'right');
+          if (direction === 'ltr') {
+            if (colIndexBefore < lastColIndex) {
+              goToCell(colIndexBefore + 1, getRowIdFromIndex(rowIndexBefore), 'right');
+            }
+          } else if (direction === 'rtl') {
+            if (colIndexBefore > firstColIndex) {
+              goToCell(colIndexBefore - 1, getRowIdFromIndex(rowIndexBefore), 'right');
+            }
           }
           break;
         }
 
         case 'ArrowLeft': {
-          if (colIndexBefore > firstColIndex) {
-            goToCell(colIndexBefore - 1, getRowIdFromIndex(rowIndexBefore));
+          if (direction === 'ltr') {
```

Comment:
We should do the same with column header navigation - it's still inverted on RTL

https://user-images.githubusercontent.com/13808724/198098211-7e1aef10-f33a-4ab9-951a-08ac664a800d.mov